posydon/binary_evol/DT/step_detached.py

```diff
@@ -1174,11 +1174,16 @@ def get_star_data(binary, star1, star2, htrack,
 
 
         if interp1d_sec is None or interp1d_pri is None:
-            # binary.event = "END"
-            binary.state += " (GridMatchingFailed)"
-            if self.verbose or self.verbose == 1:
-                print("Failed matching")
-            return
+
+            if binary.state in ['detached', 'disrupted', 'merged']:
+                binary.state += " (GridMatchingFailed)"
+                if self.verbose or self.verbose == 1:
```

Comment:
I know, you haven't change that code part, but out of curiosity: Why do we check `self.verbose` for the boolean representation and whether it is 1? Usually, if the second condition is fulfilled the first one should be as well. (There are ways in python to create objects, which would need the second condition to capture it, because the first one doesn't. But those would be a very weird object, which needs to be explicitly designed that way, no common object should behave like that.)

Reply:
I'm not sure why this is written this way. I've noticed it used in other places, like line 1505:
```
if self.verbose and self.verbose != 1:
                print("ODE solver duration: "
                      f"{t_after_ODEsolution-t_before_ODEsolution:.6g}")
```
This example in particular does not make sense to me.

According to the .ini file, this parameter should be a boolean, but it looks like the code is also allowing it to be 0/1 (presumably). Overall its usage in the detached step seems to be inconsistent throughout.
It should be simple to go through and fix this, I could do that in a different PR.